kernel/src/main/scala/cats/Eq.scala

```diff
@@ -0,0 +1,106 @@
+package cats
+
+import scala.{specialized => sp}
+
+import scala.math.Equiv
+
+/**
+ * A type class used to determine equality between 2 instances of the same
+ * type. Any 2 instances `x` and `y` are equal if `eqv(x, y)` is `true`.
+ * Moreover, `eqv` should form an equivalence relation.
+ */
+trait Eq[@sp A] extends Any with Serializable { self =>
```

Comment:
So given the binary compatibility constraints, and the fact that I think we're hoping to release cats 1.0 pretty soon now, The basic question for me is "will we get typeclassic working before we want to release cats 1.0?"

If yes, then things are easy: we can plug in simulacrum now (or not) and either way we can easily change to typeclassic without worrying too much about what exactly is produced. Ideally it wouldn't change but I wouldn't be scared of it.

If not, then I think we'll need to proceed with a bit more caution, in order to avoid reifying simulacrum's current behavior as the output that typeclassic *must* conform to.